### PR TITLE
Added cost function pointer to request.

### PR DESCRIPTION
**Pull request by scopatz**
Thanks @FlanFlanagan!

### CODE REVIEW

**Comment by scopatz** [comment]
ping!

**Comment by rwcarlsen** [comment]
Also - since Cyclus is c++11 - why not use std::function and lambdas? -
that would make it easier to include facility time-step specific state in
the cost function (i.e. with closed over vars).

File: src/request.h [hunk 1]
```diff
@@ -26,6 +28,8 @@ template <class T> class RequestPortfolio;
 template <class T>
 class Request {
  public:
+  typedef double (*cost_function_t)(boost::shared_ptr<Material>);
```
Comment (scopatz) [inline src/request.h]: Good call, I have added this now.

**Comment by scopatz** [comment]
Now uses std::function, thanks for catching that @rwcarlsen

**Comment by scopatz** [comment]
Ping again!

**Pull request merged**